File: bridgeAddon/tests/fakes/transcript.py
# nvdaMcpBridge tests -- FakeTranscript, standing in for the Transcript port.
# Copyright (C) 2026 Marlon Brandao de Sousa. GPL-2. See COPYING.txt.
#
# FAKES: domain/ports/transcript.py
#
# Records each event as a tuple so a test asserts the exact vocabulary the
# Session emitted, in order, without a filesystem. ``fail_on`` makes a named
# event raise -- the tool that proves teardown finishes (and the synth still
# restores) even when the transcript throws mid-teardown.

from __future__ import annotations

from typing import Any

from nvdaMcpBridge.domain.ports.transcript import Transcript


class FakeTranscript(Transcript):
	"""An in-memory :class:`Transcript` that records every event."""

	def __init__(self, path: str = "session.log", *, fail_on: set[str] | None = None) -> None:
		self._path = path
		self._fail_on = fail_on or set()
		self.events: list[tuple[Any, ...]] = []
		self.opened = False

	@property
	def path(self) -> str:
		return self._path

	def _record(self, name: str, *args: Any) -> None:
		self.events.append((name, *args))
		if name in self._fail_on:
			raise RuntimeError(f"transcript failing on {name}")

	def open(self) -> None:
		self.opened = True
		self._record("open")

	def session_opened(self, mode: str, synth: str) -> None:
		self._record("session_opened", mode, synth)

	def synth_swapped(self, real_synth: str) -> None:
		self._record("synth_swapped", real_synth)

	def synth_restored(self, real_synth: str) -> None:
		self._record("synth_restored", real_synth)

	def gesture(self, gesture_id: str) -> None:
		self._record("gesture", gesture_id)

	def speech(self, text: str) -> None:
		self._record("speech", text)

	def note(self, text: str) -> None:
		self._record("note", text)

	def session_closed(self, reason: str) -> None:
		self._record("session_closed", reason)

File: bridges/voiceover/Tests/Fakes/Transcript.swift
// Hand-written stateful fake for the Transcript port.

import VoiceOverBridgeDomain

public final class FakeTranscript: Transcript {
	public struct Opened: Equatable {
		public let mode: String
		public let voice: String
		public let persona: String
	}

	public var logPath: String
	public private(set) var isOpen = false
	public private(set) var opened: [Opened] = []
	public private(set) var speeches: [String] = []
	public private(set) var gestures: [String] = []
	public private(set) var typedLengths: [Int] = []
	public private(set) var announcements: [String] = []
	public private(set) var notes: [String] = []
	public private(set) var closedReasons: [String] = []

	public init(logPath: String = "/tmp/fake-session.log") {
		self.logPath = logPath
	}

	public func open() {
		isOpen = true
	}

	public func sessionOpened(mode: String, voice: String, persona: String) {
		opened.append(Opened(mode: mode, voice: voice, persona: persona))
	}

	public func speech(_ text: String) {
		speeches.append(text)
	}

	public func gesture(_ command: String) {
		gestures.append(command)
	}

	public func typed(_ length: Int) {
		typedLengths.append(length)
	}

	public func announced(_ text: String) {
		announcements.append(text)
	}

	public func note(_ text: String) {
		notes.append(text)
	}

	public func sessionClosed(reason: String) {
		closedReasons.append(reason)
		isOpen = false
	}
}
